### ISSUE TITLE
[BUG/ISSUE] Cannot get pointer to NOAA_GMD_CH4 error in GCHP 12.7.0 at c180

### ISSUE THREAD

**Issue by lizziel** [geoschem/geos-chem#219]
Running at c24, c48, and c90 with the GCHP 12.7.0 standard simulation default settings works on the Harvard compute cluster with ifort18 and OpenMPI4. However, when I increase the resolution to c180 I get the following error:
```
=====================================================================
GEOS-Chem ERROR: Cannot get pointer to NOAA_GMD_CH4!
 -> at SET_CH4 (in module GeosCore/set_global_ch4_mod.F90)
=====================================================================
```
This happens when I run GEOS-Chem 12.7.0 with the legacy GCHP framework (GCHP repository) and the experimental GCHPctm framework version 13.0.0-alpha.1 (GCHPctm repository). This implies the issue is likely in GEOS-Chem itself and is not related to external libraries.

I am investigating this issue and hope to have a fix in 12.7.1.

**Comment by lizziel** [comment]
I have verified this issue is related to the surface VMRs update that went into 12.7.0. I am able to reproduce it at any core count with using GCC 8.2, but must turn off emissions to get pass other issue https://github.com/geoschem/geos-chem/issues/227.

**Comment by lizziel** [comment]
I am able to reproduce this issue at c24 by changing the timesteps to match those used for c180.

I also figured out the problem is not an issue with NOAA_GMD_CH4 pointer. The recent surface VMRs update included a change in SET_CH4 to read NOAA_GMD_CH4 if the emissions year is after 1978, else read CMIP6 surface methane file. In either case a message was printed that NOAA_GMD_CH4 wasn't found. For this problem at c180, the issue was the emissions year used in the condition was -1, prompting read of CMIP6, but that file was not read in so no pointer was available in HEMCO. Then the incorrect error message was printed that NOAA_GMD_CH4 wasn't found.

I am now looking into why the timestep change, specifically halving the chemistry and dynamic timesteps, results in not setting the emissions year.

**Comment by christophkeller** [comment]
I think the issue has to do with the call to `SET_CH4` in gigc_chunk_mod.F90:

```
#if defined( MODEL_GEOS )
   IF ( .NOT. Input_Opt%LCH4EMIS .AND. ( DoTurb .OR. DoTend ) ) THEN
#else
   IF ( Phase /= 2 .AND. Input_Opt%ITS_A_FULLCHEM_SIM  &
         .AND. IND_('CH4','A') > 0 ) THEN
#endif
      CALL SET_CH4 ( Input_Opt%AmIRoot, Input_Opt, State_Chm, State_Diag, &
                                   State_Grid, State_Met, RC )
```

The way this is set up, GCHP will call `SET_CH4` on every time step, and this can fail if emissions are not called during the first time step (because of the way the clocks are specified).
This can be avoided by coupling the `SET_CH4` call to emissions and/or turbulence, i.e. only set the boundary CH4 conditions if it's time to do emissions or turbulence.

**Comment by lizziel** [comment]
In GEOS-Chem Classic we call SET_CH4 on every timestep if it's a fullchem simulation and CH4 is defined, hence the same call criteria here in GCHP. Within SET_CH4, setting the boundary condition only happens the first time it is called so I don't think we should need to reduce the frequency of calling it. But the first time it is called does need to line up with the first time emissions is run. It seems this bug is really that emissions isn't being run the first timestep when it should be.

**Comment by lizziel** [comment]
See also GCHP repo issue https://github.com/geoschem/gchp/issues/62.

**Comment by lizziel** [comment]
This is now fixed with 12.7.2. See https://github.com/geoschem/gchp/issues/62 for the resolution.
